### ISSUE TITLE
MGLAnnotationView return nil 

### ISSUE THREAD

**Issue by iBinh** [mapbox/mapbox-gl-native#6174]
**Platform:* iOS*
**Mapbox SDK version:*3.4.0-alpha-2*

### Steps to trigger behavior

 1. Add annotation to mapView with identifier
 2. return `MGLAnnotationView` in `-(MGLAnnotationView *)mapView:(MGLMapView *)mapView viewForAnnotation:(CustomAnnotation *)annotation;`

 3. Retrieve MGLAnnotationView with delegate ` - (nullable __kindof MGLAnnotationView *)dequeueReusableAnnotationViewWithIdentifier:(NSString *)identifier;`
`MGLAnnotationView` return nil 

@1ec5

**Comment by 1ec5** [comment]
Please post your entire implementation of `-mapView:viewForAnnotation:`. Thanks!

**Comment by iBinh** [comment]
```
   MGLAnnotationView *annotationView = [_mapBoxView dequeueReusableAnnotationViewWithIdentifier:annotation.reuseIdentifier];
        if (!annotationView) {
            annotationView = [[MGLAnnotationView alloc] initWithReuseIdentifier:annotation. annotation.reuseIdentifier];
            annotationView.frame = CGRectMake(0, 0, 40, 40);
            CGFloat hue = (CGFloat)annotation.coordinate.longitude / 100;
            annotationView.backgroundColor = [UIColor colorWithHue:hue saturation:0.5 brightness:1 alpha:1];
        }
        return annotationView;
```

**Comment by 1ec5** [comment]
Is the issue that the call to `-dequeueReusableAnnotationViewWithIdentifier:` at the top of this implementation returns `nil`? That’s expected behavior for the first time you dequeue an annotation view with a new reuse identifier. Assuming `reuseIdentifier` is a custom property you added to your annotation class, make sure the getter returns the value you expect.

**Comment by iBinh** [comment]
No, I call `dequeueReusableAnnotationViewWithIdentifier` in another method (in `mapView:didUpdateUserLocation` ), the annotation view was created and displayed on mapview, the identifier I dequeue is the same with annotation `reuseIdentifier`

**Comment by 1ec5** [comment]
`-mapView:viewForAnnotation:` needs to be called before `-mapView:didUpdateUserLocation:` in order for that to work. A view is only placed on the reuse queue at the moment `-mapView:viewForAnnotation:` returns; `-mapView:viewForAnnotation:` is only called when adding an annotation to the map.

**Comment by iBinh** [comment]
let me explain my implementation: 
I create an annotation and add it in -`mapView:didUpdateUserLocation:`
if (!customAnnotation) {
 // create and add to mapBoxView
}
// dequeue annotation view with `dequeueReusableAnnotationViewWithIdentifier`
// update position using UIView animation
is it ok?
my purpose is animate annotation view with new coordinate

**Comment by 1ec5** [comment]
If your implementation of `-mapView:didUpdateUserLocation:` is supposed to obtain the view that MGLMapView is using for the annotation you’ve just added, then you should call `-[MGLMapView viewForAnnotation:]`, passing in the annotation itself, rather than `-dequeueReusableAnnotationViewWithIdentifier:`. `-dequeueReusableAnnotationViewWithIdentifier:` is only intended for use in `-[MGLMapViewDelegate mapView:viewForAnnotation:]`.

**Comment by iBinh** [comment]
Thank you! But the annotationView does not animate, it disappear and then appear again.
Are there any ways to update annotation coordinate, when I move the mapView, the annotationView appear on mapView from  it's first location

**Comment by 1ec5** [comment]
This is probably #6203, which is being fixed in #6215.

Closing, since the original reported issue is addressed.
